source/includes/fundamentals/code-snippets/BulkWrite.java

```diff
@@ -0,0 +1,226 @@
+package docs;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.InsertOneModel;
+import com.mongodb.client.model.UpdateOneModel;
+import com.mongodb.client.model.ReplaceOneModel;
+import com.mongodb.client.model.BulkWriteOptions;
+import com.mongodb.client.model.DeleteOneModel;
+import com.mongodb.client.model.DeleteManyModel;
+
+import com.mongodb.MongoBulkWriteException;
+
+import org.bson.Document;
+
+import java.util.*;
+
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.Updates;
+import com.mongodb.client.model.WriteModel;
+
+public class BulkWrite {
+    
+    private final MongoCollection<Document> collection;
+    private final MongoClient mongoClient;
+    private final MongoDatabase database;
+
+    private BulkWrite() {
+        final String uri = System.getenv("DRIVER_REF_URI");
+
+        mongoClient = MongoClients.create(uri);
+        database = mongoClient.getDatabase("crudOps");
+        collection = database.getCollection("bulkWrite");
+    }
+
+    public static void main(String[] args) {
+        BulkWrite bulkWrite = new BulkWrite();
+        System.out.println("Ordered BulkWrite");
+        bulkWrite.setUpCollection();
+        bulkWrite.bulkWriteExample();
+        bulkWrite.preview();
+
+        System.out.println("Unordered BulkWrite");
+        bulkWrite.setUpCollection();
+        bulkWrite.bulkWriteNotOrderedExample();
+        bulkWrite.preview();
+
+        System.out.println("Insert BulkWriteException");
+        bulkWrite.setUpCollection();
+        bulkWrite.insertExceptionExample();
+
+        System.out.println("Insert");
+        bulkWrite.insertDocumentsExample();
+        bulkWrite.preview();
+
+        System.out.println("Replace");
+        bulkWrite.replaceDocumentsExample();
+        bulkWrite.preview();
+
+        System.out.println("Update");
+        bulkWrite.updateDocumentsExample();
+        bulkWrite.preview();
+
+        System.out.println("Delete");
+        bulkWrite.deleteDocumentsExample();
+        bulkWrite.preview();
+    }
+
+    
+    private void insertExceptionExample() {
+        // begin insertExceptionExample
+        try {
+            List<WriteModel<Document>> bulkOperations = new ArrayList<>();
+            
+            InsertOneModel<Document> doc3 = new InsertOneModel<>(new Document("_id", 1));
+            InsertOneModel<Document> doc4 = new InsertOneModel<>(new Document("_id", 3));
+            
+            bulkOperations.add(doc3);
+            bulkOperations.add(doc4);
+            
+            collection.bulkWrite(bulkOperations); 
+        
+        } catch (MongoBulkWriteException e){
+            collection.find().forEach(doc -> System.out.println(doc.toJson()));
+            
```

Comment:
suggestion: matching the suggestion on `bulk.txt` to remove the line logging out the documents.

```suggestion
```